### PR TITLE
Excercise 3 solutions

### PR DESCRIPTION
**Pull request by SG-55**
### 👨‍🎓👩‍🎓 Full name of the Author
Full name: Stana Gospavic

### ✅ Checklist
- [x] The submitted code is cleaned up.
- [ ] Solutions passes all tests.

### CODE REVIEW

File: src/main/kotlin/exercise3/task3/SherlockValidatesString.kt [hunk 1]
```diff
@@ -29,7 +29,26 @@ package exercise3.task3
  */
 
 internal fun isSherlockValid(s: String): String {
-    TODO("Implement me!!!")
+    val charCount = s.groupingBy { it }.eachCount()
```
Comment (vuksa) [inline src/main/kotlin/exercise3/task3/SherlockValidatesString.kt:32]: Does your solution work for the input `accc`? 

What was the idea behind `frequencies.count { it == minFreq } == 1 && minFreq == 1` check?

File: src/main/kotlin/exercise3/task1/BalancedBrackets.kt [hunk 2]
```diff
@@ -24,9 +26,32 @@ package exercise3.task1
  *
  */
 
-
 internal fun isExpressionBalanced(expression: String): Boolean {
-    TODO("Implement me!!!")
+    val stack = Stack<Char>()
```
Comment (vuksa) [inline src/main/kotlin/exercise3/task1/BalancedBrackets.kt:30]: Well done! Can you try implementing this functional way for a practice?

File: src/main/kotlin/exercise3/task3/SherlockValidatesString.kt
Comment (vuksa) [inline src/main/kotlin/exercise3/task3/SherlockValidatesString.kt:1]: It seems that the test class once again doesn't fully cover all cases, which misled you in not covering cases such as `act` input.

File: src/main/kotlin/exercise3/task4/TaxiParkTask.kt [hunk 3]
```diff
@@ -10,23 +10,30 @@ package exercise3.task4
  * Find all the drivers who performed no trips.
  */
 internal fun TaxiPark.findFakeDrivers(): Set<Driver> {
-    TODO("Implement me!!!")
+    return allDrivers - trips.map { it.driver }.toSet()
```
Comment (vuksa) [inline src/main/kotlin/exercise3/task4/TaxiParkTask.kt:13]: Well done!

File: src/main/kotlin/exercise3/task3/SherlockValidatesString.kt (hunk 1 above)
Comment (vuksa) [reply src/main/kotlin/exercise3/task3/SherlockValidatesString.kt:32]: Got it :) We can remove `a`, which gives us a correct string. :)

File: src/main/kotlin/exercise3/task1/BalancedBrackets.kt (hunk 2 above)
Comment (SG-55) [reply src/main/kotlin/exercise3/task1/BalancedBrackets.kt:30]: I'll try

File: src/main/kotlin/exercise3/task3/SherlockValidatesString.kt
Comment (SG-55) [reply src/main/kotlin/exercise3/task3/SherlockValidatesString.kt:1]: It should work for the 'act' input and cover all cases

**Pull request closed without merging**